mito-ai/mito_ai/streamlit_conversion/validate_streamlit_app.py

```
@@ -23,102 +23,97 @@
+            f.write(app_code)
+            temp_path = f.name
 
+        try:
+            # Run Streamlit test from file with UTF-8 encoding
+            app_test = AppTest.from_file(temp_path, default_timeout=30)
+            app_test.run()
+            
+            # Check for exceptions
+            if app_test.exception:
+                errors = [{'type': 'exception', 'details': exc.value, 'message': exc.message, 'stack_trace': exc.stack_trace} for exc in app_test.exception]
+                return errors
+                    
+            # Check for error messages
+            if app_test.error:
+                errors = [{'type': 'error', 'details': err.value} for err in app_test.error]
+                return errors
+            
+            return None
+        finally:
+            # Clean up the temporary file
             try:
-                # Run Streamlit test from file with UTF-8 encoding
-                app_test = AppTest.from_file(temp_path, default_timeout=30)
-                app_test.run()
-                
-                # Check for exceptions
-                if app_test.exception:
-                    errors = [{'type': 'exception', 'details': exc.value, 'message': exc.message, 'stack_trace': exc.stack_trace} for exc in app_test.exception]
-                    return errors
-                        
-                # Check for error messages
-                if app_test.error:
-                    errors = [{'type': 'error', 'details': err.value} for err in app_test.error]
-                    return errors
-                
-                return None
-            finally:
-                # Clean up the temporary file
-                try:
-                    os.unlink(temp_path)
-                except OSError:
-                    pass  # File might already be deleted
+                os.unlink(temp_path)
+            except OSError:
```

We can propagate this error as well. We can add TODOs for this if this need not be addressed immediately